### PR TITLE
Attributes should be used when fields are missing

### LINKED ISSUE #102: If a span has standard values set in attributes, but not in the Span object, the marshaller overrides them.
For example, if you set `service.name` as a attribute on a span, but don't explicitly set the `serviceName` field via the `SpanBuilder`, then the resulting `service.name` attribute is not written at all.

### PR DESCRIPTION
**Pull request by breedx-nr**
This resolves #102

### CODE REVIEW

File: telemetry_core/src/main/java/com/newrelic/telemetry/spans/json/SpanJsonTelemetryBlockWriter.java [hunk 1]
```diff
@@ -40,15 +45,34 @@ public void appendTelemetryJson(SpanBatch batch, JsonWriter jsonWriter) {
...
+
+    private final Span span;
+    private final Map<String, Object> newAttrs;
+
+    AttributesEnhancer(Span span) {
+      this.span = span;
+      this.newAttrs = new HashMap<>(span.getAttributes().asMap());
+    }
+
+    Map<String, Object> apply() {
```
Comment (jkwatson) [inline telemetry_core/src/main/java/com/newrelic/telemetry/spans/json/SpanJsonTelemetryBlockWriter.java]: why not call this `enhance`?

Comment (breedx-nr) [reply telemetry_core/src/main/java/com/newrelic/telemetry/spans/json/SpanJsonTelemetryBlockWriter.java]: Into it.

**Comment by jkwatson** [comment]
much easier to just use `putIfAbsent` for this, I think.

**Comment by breedx-nr** [comment]
Agreed.  Totally derped on the existence of that.

File: telemetry_core/src/main/java/com/newrelic/telemetry/spans/json/SpanJsonTelemetryBlockWriter.java [hunk 2]
```diff
@@ -42,10 +42,10 @@ public void appendTelemetryJson(SpanBatch batch, JsonWriter jsonWriter) {
...
   private Map<String, Object> enhanceAttributes(Span span) {
     Map<String, Object> result = new HashMap<>(span.getAttributes().asMap());
-    result.put("name", span.getName());
-    result.put("parent.id", span.getParentId());
-    result.put("duration.ms", span.getDurationMs());
-    result.put("service.name", span.getServiceName());
+    result.putIfAbsent("name", span.getName());
+    result.putIfAbsent("parent.id", span.getParentId());
+    result.putIfAbsent("duration.ms", span.getDurationMs());
+    result.putIfAbsent("service.name", span.getServiceName());
```
Comment (jkwatson) [inline telemetry_core/src/main/java/com/newrelic/telemetry/spans/json/SpanJsonTelemetryBlockWriter.java]: love it!

**Pull request merged**